web/src/buddy_list.js

```diff
@@ -278,6 +278,7 @@ export class BuddyList extends BuddyListConf {
                 header_text,
                 user_count: get_formatted_sub_count(subscriber_count),
                 toggle_class: "toggle-users-matching-view",
+                is_collapsed: $("#buddy-list-users-matching-view").hasClass("collapsed"),
```

Comment:
Is there a reason to read this from the DOM rather than making a global variable to keep track of that state? I guess it doesn't really matter, but it feels like this module should be keeping track of whether that is collapsed for itself, rather than using the DOM as its data storage.

Reply:
My general preference is to avoid having multiple sources of truth, i.e. a variable could end up out of sync with the DOM. The main reason I'd keep a variable is if there's concerns about performance or something, which I don't have here. I'm open to hearing more reasons to change it though!

Reply:
I see the global variable as a dedicated place to store "Should this be collapsed" and the DOM state as "is this currently collapsed?" 

The approach you have here is very fragile, in that you're reading this state after code earlier has emptied the adjacent containers, and it's not obvious to me that this code as written even works -- didn't we just delete from the DOM the thing we're reading from at the top of this function?

Maybe it's in a slightly different bucket, but even if this turns out to work as written, it's fragile and hard to verify. What is not fragile and hard to verify is having a single source of truth, a variable that tracks the state, and rendering logic that aligns with it.

Reply:
Okay, I see. Thanks for explaining.